lxd/network_allocations.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	clusterRequest "github.com/canonical/lxd/lxd/cluster/request"
+	"github.com/canonical/lxd/lxd/db"
+	dbCluster "github.com/canonical/lxd/lxd/db/cluster"
+	"github.com/canonical/lxd/lxd/network"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/version"
+)
+
+var networkAllocationsCmd = APIEndpoint{
+	Path: "network-allocations",
+
+	Get: APIEndpointAction{Handler: networkAllocationsGet, AccessHandler: allowProjectPermission("networks", "view")},
+}
+
+// API endpoints.
+
+// swagger:operation GET /1.0/network-allocations network-allocations network_allocations_get
+//
+//  Get the network allocations in use (`network`, `network-forward` and `load-balancer` and `instance`)
+//
+//  Returns a list of network allocations in use by a LXD deployment.
+//
+//  ---
+//  produces:
+//    - application/json
+//  parameters:
+//    - in: query
+//      name: project
+//      description: Project name
+//      type: string
+//      example: default
+//    - in: query
+//      name: all-projects
+//      description: Retrieve entities from all projects
+//      type: boolean
+//  responses:
+//    "200":
+//      description: API endpoints
+//      schema:
+//        type: object
+//        properties:
+//          type:
+//            type: string
+//            description: Response type
+//            example: sync
+//          status:
+//            type: string
+//            description: Status description
+//            example: Success
+//          status_code:
+//            type: integer
+//            description: Status code
+//            example: 200
+//          metadata:
+//            type: array
+//            description: List of network allocations used by a consuming entity
+//            items:
+//              $ref: "#/definitions/NetworkAllocations"
+//    "403":
+//      $ref: "#/responses/Forbidden"
+//    "500":
+//      $ref: "#/responses/InternalServerError"
+func networkAllocationsGet(d *Daemon, r *http.Request) response.Response {
+	var err error
+	projectName := queryParam(r, "project")
```

Comment:
I think this should probably be:

```go
	projectName, _, err := project.NetworkProject(s.DB.Cluster, projectParam(r))
	if err != nil {
		return response.SmartError(err)
	}
```

So that if your in a project that doesn't `features.networks` enabled, then doing `lxc network list-allocations` will get you usage in the default project.